tests/integration/test_external_agent.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-k8s-operator with ingress."""
+
+import typing
+
+import pytest
+from juju.application import Application
+
+import state
+from charm import AGENT_DISCOVERY_INGRESS_RELATION_NAME
+
+
+# This will only work on microk8s !!
+@pytest.mark.abort_on_fail
+async def test_agent_discovery_ingress_integration(
+    application: Application,
+    traefik_application_and_unit_ip: typing.Tuple[Application, str],
+    external_hostname: str,
+    jenkins_machine_agents: Application,
+):
+    """
+    arrange: deploy the Jenkins charm, ingress, and a machine agent.
+    act: integrate the charms with each other and update the machine agent's dns record.
+    assert: All units should be in active status.
```

Comment:
I'm wondering if a check to test if the external agent is correctly registered would be beneficial here :D

Reply:
I think in this test we actually run a job on the machine agent and check if the job passes, so I think that should suffice. what do you think?

Reply:
Thanks for letting me know!